### PR TITLE
feat/game: (mostly) fully featured poker core code

### PR DESCRIPTION
**Pull request by solunian**
## DONE
- poker core working, full games can be run
- connected to db and admin api routes (for the most part, still some kinks to work out)
- connected to bots and format is comprehensive and handles errors "gracefully"
- GameState schema finalized and syncs to `tables` table in the db (finally, like 3 iterations)

## TODO
- providing skeleton code for users?
- needs testing, but merge into main so that stuff can be flushed out
- probably needs more documentation and explanations for aspects of code (still hella comments)
- feat: last_change

### CODE REVIEW

**Review by ken-tummada — APPROVED** [review]
lgtm

**Review by smmzhu — APPROVED** [review]
stamping to unblock, but few things to note:
- i did not test this lmao, please do rigorous testing of the core logic. we don't want some weird edge case to mess up who should be the winner on actual game day. depending on how much time you have (and also if you are trying to make this a resume winner), you may want to include some testing — something like pytest could cover edge cases
- all comments aren't functional gaps in your code, more like code smells and clerical stuff
-

File: src/core/hand.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        straight_flush = Hand.straight(all_flush) if all_flush is not None else None
+        if straight_flush is not None:
+            self.cards = straight_flush
+            self.type = "straight flush"
+            return
+
+        four_of_a_kind = Hand.four_of_a_kind(cards, rank_occurences)
+        if four_of_a_kind is not None:
+            self.cards = four_of_a_kind
+            self.type = "four of a kind"
```
[the remark this answers is not in the document]
Comment (ken-tummada) [reply src/core/hand.py]: agree, changed.
